pkg/pipeline/PipelineBuilder.go

```diff
@@ -106,72 +106,149 @@ func GetDeploymentServiceTypeConfig() (*DeploymentServiceTypeConfig, error) {
 	CreateApp(request *bean.CreateAppDTO) (*bean.CreateAppDTO, error)
-	CreateMaterialsForApp(request *bean.CreateMaterialDTO) (*bean.CreateMaterialDTO, error)
-	UpdateMaterialsForApp(request *bean.UpdateMaterialDTO) (*bean.UpdateMaterialDTO, error)
-	DeleteMaterial(request *bean.UpdateMaterialDTO) error
+	//DeleteApp : Delegating the request to ciCdPipelineOrchestrator for application deletion.
 	DeleteApp(appId int, userId int32) error
+	//GetApp : Gets Application along with Git materials for given appId.
+	GetApp(appId int) (application *bean.CreateAppDTO, err error)
+	//FindByIds : Find applications by given IDs, delegating the request to the appRepository.
+	FindByIds(ids []*int) ([]*AppBean, error)
+	//GetAppList : Retrieve and return a list of applications.
+	GetAppList() ([]AppBean, error)
+	//FindAllMatchesByAppName : Find and return applications matching the given name and type.
+	FindAllMatchesByAppName(appName string, appType helper.AppType) ([]*AppBean, error)
+	//GetAppListForEnvironment : Retrieve and return authorized applications based on the provided request.
+	GetAppListForEnvironment(request resourceGroup2.ResourceGroupingRequest) ([]*AppBean, error)
+	//FindAppsByTeamId : Retrieve applications associated with the specified team ID , delegating the request to appRepository.
+	FindAppsByTeamId(teamId int) ([]*AppBean, error)
+	//FindAppsByTeamName : Retrieve applications associated with the specified team Name , delegating the request to appRepository.
+	FindAppsByTeamName(teamName string) ([]AppBean, error)
+
+	//function related to ciPipeline
+	//GetCiPipeline : Retrieve ciPipeline or jobPipeline for given appId
 	GetCiPipeline(appId int) (ciConfig *bean.CiConfigRequest, err error)
+	//GetTriggerViewCiPipeline :
 	GetTriggerViewCiPipeline(appId int) (*bean.TriggerViewCiConfig, error)
+	//GetExternalCi : Retrieve externalCi for given appId
 	GetExternalCi(appId int) (ciConfig []*bean.ExternalCiConfig, err error)
+	//GetExternalCiById : Retrieve externalCi for given appId and externalCiId
 	GetExternalCiById(appId int, externalCiId int) (ciConfig *bean.ExternalCiConfig, err error)
+	//UpdateCiTemplate : Update the CI template configuration, ensuring consistency with the requested changes,
+	//and return the updated configuration
 	UpdateCiTemplate(updateRequest *bean.CiConfigRequest) (*bean.CiConfigRequest, error)
+	//PatchCiPipeline : Handle CI pipeline patch requests, making necessary changes to the configuration and returning the updated version.
+	//Performs Create ,Update and Delete operation.
 	PatchCiPipeline(request *bean.CiPatchRequest) (ciConfig *bean.CiConfigRequest, err error)
+	//CreateCiPipeline : Create a CI pipeline based on the provided configuration request.
+	CreateCiPipeline(createRequest *bean.CiConfigRequest) (*bean.PipelineCreateResponse, error)
+	//GetCiPipelineMin : lists minimum detail of ciPipelines for given appId and envIds
+	GetCiPipelineMin(appId int, envIds []int) ([]*bean.CiPipelineMin, error)
```

Comment:
what is minimum? how is it different from GetCiPipeline? when to use that when to use this?